src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -130,5 +133,32 @@ protected override Task WriteResponseHeadersAsyncCore(Metadata responseHeaders)
 
             return HttpContext.Response.Body.FlushAsync();
         }
+		
+        public void Initialize()
+        {
+            var timeout = GrpcProtocolHelpers.GetTimeout(HttpContext);
+
+            if (timeout != TimeSpan.MaxValue)
+            {
+                // CancellationTokenSource does not support greater than int.MaxValue milliseconds
+                if (timeout.TotalMilliseconds > int.MaxValue)
+                {
+                    throw new InvalidOperationException("A timeout greater than 2147483647 milliseconds is not supported.");
+                }
+
+                _deadline = Clock.UtcNow.Add(timeout);
+                _cts = new CancellationTokenSource(timeout);
+            }
+            else
+            {
+                _deadline = DateTime.MaxValue;
+                _cts = new CancellationTokenSource();
```

Comment:
in this case, how would the _cts be cancelled?

Reply:
Client could cancel.